server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationAllocationIT.java

```diff
@@ -0,0 +1,929 @@
+
+    static boolean addRemote = false;
+    private final ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+    private Client client;
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        if (segmentRepoPath == null || translogRepoPath == null) {
+            segmentRepoPath = randomRepoPath().toAbsolutePath();
+            translogRepoPath = randomRepoPath().toAbsolutePath();
+        }
+        if (addRemote) {
+            logger.info("Adding remote_store_enabled node");
+            return Settings.builder()
+                .put(super.nodeSettings(nodeOrdinal))
+                .put(remoteStoreClusterSettings(REPOSITORY_NAME, segmentRepoPath, REPOSITORY_2_NAME, translogRepoPath))
+                .put("discovery.initial_state_timeout", "500ms")
+                .build();
+        } else {
+            logger.info("Adding non_remote_store_enabled node");
+            return Settings.builder().put(super.nodeSettings(nodeOrdinal)).put("discovery.initial_state_timeout", "500ms").build();
+        }
+    }
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.REMOTE_STORE_MIGRATION_EXPERIMENTAL, "true").build();
+    }
+
+    @Override
+    protected boolean forbidPrivateIndexSettings() {
+        return false;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(InternalOrPrivateSettingsPlugin.class);
+    }
+
+    // tests for primary shard copy allocation with REMOTE_STORE direction
+
+    public void testDontAllocateNewPrimaryShardOnNonRemoteNodeForRemoteStoreDirection() throws Exception {
+        logger.info(" --> initialize cluster");
+        initializeCluster();
+
+        logger.info(" --> add non-remote node");
+        addRemote = false;
+        String nonRemoteNodeName = internalCluster().startNode();
+        internalCluster().validateClusterFormed();
+        DiscoveryNode nonRemoteNode = assertNodeInCluster(nonRemoteNodeName);
+        assertFalse(nonRemoteNode.isRemoteStoreNode());
+
+        logger.info(" --> set remote_store direction");
+        setDirection(REMOTE_STORE_DIRECTION);
+
+        logger.info(" --> verify expected decision for allocating a new primary shard on a non-remote node");
+
+        boolean isRemoteStoreBackedIndex = randomBoolean();
+        if (isRemoteStoreBackedIndex) {
+            // to create the repositories
+            setClusterMode(MIXED_MODE);
+            addRemote = true;
+            String remoteNodeName = internalCluster().startNode();
+            internalCluster().validateClusterFormed();
+            DiscoveryNode remoteNode = assertNodeInCluster(remoteNodeName);
+            assertTrue(remoteNode.isRemoteStoreNode());
+            setClusterMode(STRICT_MODE);
+        }
+        boolean isMixedMode = randomBoolean();
```

Comment:
Should we rearrange the code block into this format?

```
if(isMixedMode) {

} else {

}
```
Might need to do a bit of duplicacy. Though will be more readable in nature.

Reply:
Moved `strict` mode decisions into separate tests.